src/dashboard/overview/ServiceDetails.tsx

```diff
@@ -1,49 +1,146 @@
+    color: "#000000",
+    textDecoration: "underline",
+  },
+  dialogTitleContainer: (props: ServiceDetailsProps) => ({
+    backgroundImage: props.titleBackground,
+    padding: "8px",
+  }),
+  dialogTitleContent: {
+    display: "flex",
+    alignItems: "baseline",
+    padding: "5px 15px",
+  },
+  logsAndCloseContainer: {
+    display: "flex",
+    marginLeft: "auto",
+    alignItems: "center",
+  },
+  dialogContent: {
+    padding: 0,
+  },
+}));
 
 const ServiceDetails = (props: ServiceDetailsProps): ReactElement => {
-  const { status, handleClose } = props;
-  const classes = useStyles();
+  const { status, handleClose, statusIcon } = props;
+  const classes = useStyles(props);
+  const errorMsgOpenSubject = new Subject<boolean>();
 
-  const title = `General ${status.service} info`;
+  const isDownloadLogsEnabled = (status: Status): boolean => {
+    return (
+      !status.status.includes("light mode") && status.status !== "Disabled"
+    );
+  };
+
+  const downloadLogs = (serviceName: string, handleError: () => void): void => {
+    api.logs$(serviceName).subscribe({
+      next: (resp: string) => {
+        const blob = new Blob([resp]);
+        const url = URL.createObjectURL(blob);
+        const anchor = Object.assign(document.createElement("a"), {
+          href: url,
+          download: `${serviceName}_${formatDateTimeForFilename(
+            new Date()
+          )}.log`,
+          style: { display: "none" },
+        });
+        anchor.click();
+      },
+      error: handleError,
+    });
+  };
 
   return (
-    <Dialog
-      open
-      handleClose={handleClose}
-      classes={{
-        paper: classes.dialog,
-      }}
-      headerVisible
-      closeIconVisible
-      title={title}
-      content={
-        <ServiceDetailsContent status={status} closeDetails={handleClose} />
-      }
-    />
+    <div>
+      <Dialog
+        open
+        onClose={handleClose}
+        classes={{
+          paper: classes.dialog,
+        }}
+      >
+        <DialogTitle className={classes.dialogTitleContainer}>
+          <div className={classes.dialogTitleContent}>
+            <span
+              className={classes.dialogTitle}
+            >{`General ${status.service} info`}</span>
+
+            <div className={classes.logsAndCloseContainer}>
+              {isDownloadLogsEnabled(status) && (
+                <div>
+                  <ButtonBase
+                    className={classes.downloadIconContainer}
+                    onClick={() =>
+                      downloadLogs(status.service, () =>
+                        errorMsgOpenSubject?.next(true)
+                      )
+                    }
+                  >
+                    <img src={downloadIcon} alt="download logs" />
+                    Download Logs
+                  </ButtonBase>
+                </div>
+              )}
+              <ButtonBase onClick={handleClose}>
+                <img src={expandIcon} alt="expand" />
```

Comment:
Instead of the expand icon, there should be the "opposite" icon that indicates the collapse.

Reply:
Done.